devicetypes/smartthings/arrival-sensor-ha.src/arrival-sensor-ha.groovy

```diff
@@ -169,13 +262,18 @@ private stopTimer() {
     log.debug "Stopping periodic timer"
     // Always unschedule to handle the case where the DTH was running in the cloud and is now running locally
     unschedule("checkPresenceCallback", [forceForLocallyExecuting: true])
+    state.gsensor = 0
 }
 
 def checkPresenceCallback() {
-    def timeSinceLastCheckin = (now() - state.lastCheckin ?: 0) / 1000
+	def timeSinceLastCheckin = (now() - state.lastCheckin ?: 0) / 1000
     def theCheckInterval = (checkInterval ? checkInterval as int : 2) * 60
     log.debug "Sensor checked in ${timeSinceLastCheckin} seconds ago"
     if (timeSinceLastCheckin >= theCheckInterval) {
-        handlePresenceEvent(false)
+    	if (device.getDataValue("manufacturer") == "Vision") {
```

Comment:
Since You are using this condition in many places You may add a separate method that returns true/false and use it in Your code, 
for example 
`isVision () {
   return device.getDataValue("manufacturer") == "Vision"
}`

Reply:
OK! I will add a method to determine the manufacturer.